critwall/src/intTest/java/org/opencadc/critwall/FileSyncJobTest.java

```diff
@@ -0,0 +1,213 @@
+            Artifact a = artifactIterator.next();
+            log.debug("deleting test uri: " + a.getURI() + " ID: " + a.getID());
+            dao.delete(a.getID());
+        }
+    }
+
+    @Before
+    public void init_cleanup() throws Exception {
+        // will have to get all artifacts from the table, iterate over them and
+        // delete. Will be completely destructive, but so be it!
+        log.debug("cleaning stored artifacts...");
+        Iterator<Artifact> storedArtifacts = dao.storedIterator(null);
+        wipe_clean(storedArtifacts);
+
+        log.debug("cleaning unstored artifacts...");
+        Iterator<Artifact> unstoredArtifacts = dao.unstoredIterator(null);
+        wipe_clean(unstoredArtifacts);
+
+    }
+
+
+    @Test
+    public void testJob() throws Exception {
+
+        try {
+            BucketMode bucketMode = FileSystemStorageAdapter.BucketMode.URI;
+            log.info("testJob(" + bucketMode + ") - start");
+
+            // Create the test directory the fs storage adapter needs
+            String testDir = TEST_ROOT + File.separator + "testJob-" + bucketMode;
+            Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwxrwxrw-");
+            FileAttribute<Set<PosixFilePermission>> attr = PosixFilePermissions.asFileAttribute(perms);
+            Files.createDirectories(Paths.get(testDir), attr);
+
+            URI artifactID = new URI(TEST_ARTIFACT_URI);
+            URI resourceID = new URI(TEST_RESOURCE_ID);
+
+            // Set up an Artifact in the database to start.
+            Artifact artifactToUpdate = new Artifact(artifactID,
+                new URI("md5:646d3c548ffb98244a0fc52b60556082"), new Date(), //storageMeta.contentLastModified,
+                1008000L);
+
+            log.debug("setting test artifact: ");
+            dao.put(artifactToUpdate);
+            //verify something is there
+            log.debug("ID of test artifact" + artifactToUpdate.getID());
+            Artifact testArtifact = dao.get(artifactToUpdate.getID());
+            Assert.assertNotNull(testArtifact);
+            log.debug("test artifact stored");
+
+            FileSystemStorageAdapter sa = new FileSystemStorageAdapter(testDir, bucketMode, 1);
+            FileSyncJob fsj = new FileSyncJob(artifactID, resourceID, sa, dao);
+            fsj.run();
+
+            // assert that job succeeded
+            // Get the artifact by URI again
+            Artifact storedArtifact = dao.get(artifactID);
+
+            // check that artifact in db has storage location
+            Assert.assertNotNull(storedArtifact.storageLocation);
+
+            StorageLocation expected = new StorageLocation(new URI("ad:IRIS/I212B2H0.fits"));
+            expected.storageBucket = "ad:IRIS";
+            Assert.assertEquals(expected, storedArtifact.storageLocation);
+
+            // check for file on disk
+            ByteArrayOutputStream dest = new ByteArrayOutputStream();
```

Comment:
If line 177 is correct, this is ~1MiB of ram. Certainly modest and probably unnoticeable... except if grabage collection doesn't happen right away or other things we don't know/care about do happen.
At least make the intent to discard clear by calling dest.reset() after the get.

If you didn't want to worry about the file size, `new FileOutputStream("/dev/null")` would work and the intent would be clear :-)

Reply:
went for /dev/null - done